internal/metastore/kv/rootcoord/kv_catalog.go

```diff
@@ -1439,6 +1466,113 @@ func (kc *Catalog) RestoreRBAC(ctx context.Context, tenant string, meta *milvusp
 	return err
 }
 
+func (kc *Catalog) GetPrivilegeGroup(ctx context.Context, groupName string) ([]*milvuspb.PrivilegeEntity, error) {
+	k := BuildPrivilegeGroupkey(groupName)
+	val, err := kc.Txn.Load(k)
+	if err != nil {
+		if errors.Is(err, merr.ErrIoKeyNotFound) {
+			return nil, fmt.Errorf("privilege group [%s] does not exist", groupName)
+		}
+		log.Error("failed to load privilege group", zap.String("group", groupName), zap.Error(err))
+		return nil, err
+	}
+	privGroupInfo := &milvuspb.PrivilegeGroupInfo{}
+	err = proto.Unmarshal([]byte(val), privGroupInfo)
+	if err != nil {
+		log.Error("failed to unmarshal privilege group info", zap.Error(err))
+		return nil, err
+	}
+	return privGroupInfo.Privileges, nil
+}
+
+func (kc *Catalog) DropPrivilegeGroup(ctx context.Context, groupName string) error {
+	// check if the group is used by any role
+	roles, err := kc.ListRole(ctx, util.DefaultTenant, nil, false)
+	if err != nil {
+		return err
+	}
+
+	roleEntity := lo.FilterMap(roles, func(entity *milvuspb.RoleResult, _ int) (*milvuspb.RoleEntity, bool) {
+		roleName := entity.GetRole().GetName()
+		if roleName == util.RoleAdmin || roleName == util.RolePublic {
+			return nil, false
+		}
+
+		return entity.GetRole(), true
+	})
+
+	for _, role := range roleEntity {
+		grants, err := kc.ListGrant(ctx, util.DefaultTenant, &milvuspb.GrantEntity{
+			Role:   role,
+			DbName: util.AnyWord,
+		})
+		if err != nil {
+			return err
+		}
+		for _, grant := range grants {
+			if grant.Grantor.Privilege.Name == groupName {
+				return errors.Newf("privilege group [%s] is used by role [%s], Use REVOKE API to revoke it first", groupName, role.GetName())
+			}
+		}
+	}
+
+	k := BuildPrivilegeGroupkey(groupName)
+	err = kc.Txn.Remove(k)
+	if err != nil {
+		log.Warn("fail to drop privilege group", zap.String("key", k), zap.Error(err))
+		return err
+	}
+	return nil
+}
+
+func (kc *Catalog) AlterPrivilegeGroup(ctx context.Context, groupName string, privileges []*milvuspb.PrivilegeEntity) error {
+	k := BuildPrivilegeGroupkey(groupName)
+	groupInfo := &milvuspb.PrivilegeGroupInfo{
+		GroupName:  groupName,
+		Privileges: privileges,
+	}
+	v, err := proto.Marshal(groupInfo)
+	if err != nil {
+		log.Error("failed to marshal privilege group info", zap.Error(err))
+		return err
+	}
+	hasKey, err := kc.Txn.Has(k)
+	if err != nil {
```

Comment:
`CreatePrivilegeGroup` in root_coord.go also call this func, but this func return err if resource group not exist. refactor this func to `SavePrivilegeGroup(ctx context.Context, data *milvus.PrivilegeGroupInfo)`, do all check logic in root_coord.go